src/components/hooks/useActivityStream.js

```diff
@@ -1,5 +1,21 @@
-import {useState, useEffect} from 'react';
-import {merge} from 'rxjs';
+import {useEffect, useReducer} from 'react';
+
+const APPEND_ACTIVITIES = 'append_activities';
+
+function reducer(activityIDs, action) {
```

Comment:
Can you add a docstring here?

```
...
Returns a new array of activityIDs based on the given action. If no action is passed, it will return the same activityIDs array without any changes.
...
```
or something like that